src/main.cpp

```diff
@@ -8614,21 +8614,20 @@ void ZeroOutResearcherTotals(std::string cpid)
 
 bool LoadAdminMessages(bool bFullTableScan, std::string& out_errors)
 {
-    int nMaxDepth = nBestHeight;
-    int nMinDepth = fTestNet ? 1 : 164618;
-    nMinDepth = pindexBest->nHeight - (BLOCKS_PER_DAY*30*12);
-    if (nMinDepth < 2) nMinDepth=2;
-    if (!bFullTableScan) nMinDepth = nMaxDepth-6;
-    if (nMaxDepth < nMinDepth) return false;
-    CBlockIndex* pindex = blockFinder.FindByHeight(nMinDepth);
-    // These are memorized consecutively in order from oldest to newest
+    // Find starting block. On full table scan we want to scan 6 months back.
+    // On a shallow scan we can limit to 6 blocks back.
+    CBlockIndex* pindex = bFullTableScan
+            ? blockFinder.FindByTime(pindexBest->nTime - MaxBeaconAge())
```

Comment:
> and stops at the block which is not older than \p time or the youngest block if it is older than \p time

Change the definition so it always returns "the highest block that is older than `time`". With the current definition of `FindByTime` it may start off-by-one.

Reply:
Will do.